base/log_level.go

```diff
@@ -0,0 +1,60 @@
+package base
+
+import (
+	"errors"
+	"fmt"
+	"sync/atomic"
+)
+
+// LogLevel is used to represent a log level.
+type LogLevel uint32
+
+const (
+	// LEVEL_NONE disables all logging
+	LEVEL_NONE LogLevel = iota
+	// LEVEL_ERROR enables only error logging.
+	LEVEL_ERROR
+	// LEVEL_WARN enables warn, and error logging.
+	LEVEL_WARN
+	// LEVEL_INFO enables info, warn, and error logging.
+	LEVEL_INFO
+	// LEVEL_DEBUG enables all logging.
+	LEVEL_DEBUG
```

Comment:
We definitely don't apply this consistently in our code yet, but I'm leaning toward sticking to mixed caps for constants, based on https://github.com/golang/go/wiki/CodeReviewComments#mixed-caps

Reply:
Changed